library/src/main/java/org/xmtp/android/library/Contacts.kt

```diff
@@ -125,80 +165,95 @@ class ConsentList(
-                        when (entry.consentType) {
-                            ConsentState.ALLOWED ->
-                                it.setAllowInboxId(
-                                    PrivatePreferencesAction.AllowInboxId.newBuilder()
-                                        .addInboxIds(entry.value),
-                                )
-
-                            ConsentState.DENIED ->
-                                it.setDenyInboxId(
-                                    PrivatePreferencesAction.DenyInboxId.newBuilder()
-                                        .addInboxIds(entry.value),
-                                )
-
-                            ConsentState.UNKNOWN -> it.clearMessageType()
+                        EntryType.INBOX_ID -> {
+                            when (entry.consentType) {
+                                ConsentState.ALLOWED ->
+                                    it.setAllowInboxId(
+                                        PrivatePreferencesAction.AllowInboxId.newBuilder()
+                                            .addInboxIds(entry.value),
+                                    )
+
+                                ConsentState.DENIED ->
+                                    it.setDenyInboxId(
+                                        PrivatePreferencesAction.DenyInboxId.newBuilder()
+                                            .addInboxIds(entry.value),
+                                    )
+
+                                ConsentState.UNKNOWN -> it.clearMessageType()
+                            }
                         }
                     }
                 }
-            }
-        }.build()
+            }.build()
+
+            val message =
+                uniffi.xmtpv3.userPreferencesEncrypt(
+                    publicKey.toByteArray(),
+                    privateKey.toByteArray(),
+                    payload.toByteArray(),
+                )
 
-        val message =
-            uniffi.xmtpv3.userPreferencesEncrypt(
-                publicKey.toByteArray(),
-                privateKey.toByteArray(),
-                payload.toByteArray(),
+            val envelope = EnvelopeBuilder.buildFromTopic(
+                Topic.preferenceList(identifier),
+                Date(),
+                ByteArray(message.size) { message[it] },
             )
 
-        val envelope = EnvelopeBuilder.buildFromTopic(
-            Topic.preferenceList(identifier),
-            Date(),
-            ByteArray(message.size) { message[it] },
-        )
+            client.publish(listOf(envelope))
+        }
+    }
 
-        client.publish(listOf(envelope))
+    suspend fun setV3ConsentState(entries: List<ConsentListEntry>) {
+        entries.forEach {
+            client.v3Client?.setConsentState(
+                ConsentState.toFfiConsentState(it.consentType),
+                EntryType.toFfiConsentEntityType(it.entryType),
+                it.value
+            )
+        }
```

Comment:
For performance reasons in this interim period of V2 and V3 we might want to add a FFibinding for setting a list so we dont have to iterate.